### PR TITLE
Use design system Tags component on Curriculum Catalog

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Switches to using the `Tags` component over the custom -- and extremely similar -- version in the curriculum catalog. The design system `Tags` component was released in December 2023, after the curriculum catalog was launched, which is presumably why this wasn't originally used.

I'm doing this now as I had some issues when trying to create a wide view of a curriculum catalog card for the congrats page, where the topic tokens had some broken styling. This seemed to fix it and, in general seemed like a good cleanup.

What it looks like now (same as before):
Screenshot 2024-04-19 at 12 17 45 PM

## Warning!!

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2024. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/componentLibrary/tags/_Tag.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ const Tag: React.FunctionComponent<TagProps> = ({
       )}
     >
       <div
+        tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
```
Comment (bethanyaconnor) [inline apps/src/componentLibrary/tags/_Tag.tsx:75]: Definitely open to feedback here but I needed this to maintain parity for keyboard users being able to tab over this component and see the tool tip. If there's a better way of doing this or if this should be conditional based on a prop, let me know!

Comment (kelbyhawn) [reply apps/src/componentLibrary/tags/_Tag.tsx:75]: Using tooltips on non-interactive elements is discouraged, but this seems to be the best solution without redesigning this part of the card! 😅

**Review by kelbyhawn — APPROVED** [review]
This looks great! I answered your question re: the `tabindex`, but maybe @hannahbergam has thoughts?

File: apps/src/componentLibrary/tags/_Tag.tsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/componentLibrary/tags/_Tag.tsx:75]: Thanks for updating this!
Yeah, adding tabIndex might not be a best solution overall, but is probably the best solution in current situation.

Once `Tooltip` component will be ready I'll probably be coming back to this component and updating it, will also take a fresh look on `Tags` overall and maybe rework it slightly

Comment (levadadenys) [reply apps/src/componentLibrary/tags/_Tag.tsx:75]: nit: making changes into Design System components also requires manually update the corresponding [CHANGELOG.md](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/componentLibrary/tags/CHANGELOG.md), please add some small description there (`0.2.0` should be a good version for this update)

**Review by levadadenys — COMMENTED** [review]
Thanks a lot for this update!
LGTM overall, just two comments:

1. Is related to Changelog (see comment)
2. `Tags` , not `Tips`. 😄  Please update the Pr heading to avoid possible future misunderstandings

**Comment by bethanyaconnor** [comment]
> Tags , not Tips. 😄 Please update the Pr heading to avoid possible future misunderstandings

🤦‍♀️ wow, that's a bad one. I updated the title and description!

File: apps/src/componentLibrary/tags/_Tag.tsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/componentLibrary/tags/_Tag.tsx:75]: done!

**Review by levadadenys — APPROVED** [review]
LGTM! Thanks again for updating this!

**Pull request merged**